### PR TITLE
Images #2 - Allow users to directly upload image files

### PR DESCRIPTION
**Pull request by Meg1456**
Working code.

### CODE REVIEW

**Comment by Meg1456** [comment]
The feature is working, but I am facing a bit of a strange issue. I need to do a hard refresh in the web page before I can see the message (i.e. I click submit and the message vanishes. Only when I refresh can I see the message with the image listed below). Is this normal behaviour? Is there something I can do to fix this?

File: .factorypath [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<factorypath>
```
Comment (potay) [inline .factorypath:1]: what's this file? is this expected?

File: src/main/webapp/js/user-page-loader.js [hunk 2]
```diff
@@ -40,8 +40,15 @@ function showMessageFormIfViewingSelf() {
       .then((loginStatus) => {
         if (loginStatus.isLoggedIn &&
             loginStatus.username == parameterUsername) {
-          const messageForm = document.getElementById('message-form');
-          messageForm.classList.remove('hidden');
+          fetch('/blobstore-upload-url')
+            .then((response) => {
```
Comment (potay) [inline src/main/webapp/js/user-page-loader.js]:
```suggestion
            .then((response) => response.text())
```

**Comment by potay** [comment]
> The feature is working, but I am facing a bit of a strange issue. I need to do a hard refresh in the web page before I can see the message (i.e. I click submit and the message vanishes. Only when I refresh can I see the message with the image listed below). Is this normal behaviour? Is there something I can do to fix this?

This is probably because when you submit the message, it gets uploaded on the backend (java), but the frontend (javascript) never refreshes and loads the new message that has been saved.

If that's not the case, and your frontend IS loading new messages after one has been saved, another possible reason is that your frontend is loading new messages too quickly, before the new message could be saved. So the message list it gets back from the server does not include the new message (because the new message is still in the process of being saved).

**Comment by Meg1456** [comment]
> > The feature is working, but I am facing a bit of a strange issue. I need to do a hard refresh in the web page before I can see the message (i.e. I click submit and the message vanishes. Only when I refresh can I see the message with the image listed below). Is this normal behaviour? Is there something I can do to fix this?
> 
> This is probably because when you submit the message, it gets uploaded on the backend (java), but the frontend (javascript) never refreshes and loads the new message that has been saved.
> 
> If that's not the case, and your frontend IS loading new messages after one has been saved, another possible reason is that your frontend is loading new messages too quickly, before the new message could be saved. So the message list it gets back from the server does not include the new message (because the new message is still in the process of being saved).

@potay Ohh. Can you please guide me on how I can fix this? I am unable to recognise the reason for the error in my code..

**Review by Reine0017 — COMMENTED** [review]
Hi Meg, I think the extra files (probably) got added because the git pull origin wasn't done when you're on master branch? (I just learnt that I've been making this mistake lol). Also, may I ask if you'd think it be ready to be included in the prototype?

**Comment by Meg1456** [comment]
> Hi Meg, I think the extra files (probably) got added because the git pull origin wasn't done when you're on master branch? (I just learnt that I've been making this mistake lol). Also, may I ask if you'd think it be ready to be included in the prototype?

Ohh ok. Yes, we can include this in the prototype. The feature is completed and working as well

**Comment by Meg1456** [comment]
Resolved merge conflict. Ready to merge.

**Review by Reine0017 — APPROVED** [review]
looks good!

**Pull request merged**